crates/router/src/connector/bluesnap.rs

```diff
@@ -1046,7 +1046,7 @@ impl api::IncomingWebhook for Bluesnap {
         Ok(match details.transaction_type.as_str() {
             "DECLINE" | "CC_CHARGE_FAILED" => api::IncomingWebhookEvent::PaymentIntentFailure,
             "CHARGE" => api::IncomingWebhookEvent::PaymentIntentSuccess,
-            _ => Err(errors::ConnectorError::WebhookEventTypeNotFound).into_report()?,
+            "UNKNOWN" => api::IncomingWebhookEvent::EventNotSupported,
```

Comment:
This is not sound. Please create a new enum type that models `DECLINE`, `CC_CHARGE_FAILED`, and other webhook events and use that enum here instead of directly matching on the string.